Remove the equality operator of `ExitCode`

sphuber opened the pull request:
Fixes #3939 

The operator implementation assumed that `other` would always be an
instance of `ExitCode` as well and so therefore was guaranteed to
contain the same attributes. However, comparing it to a bare tuple would
also invoke the `ExitCode.__eq__` method and would raise and
`AttributeError` as a result. The desired behavior for the equality
operator is that it returns True for any other tuple of the same length
and content. Since this is exactly what is implemented by the tuple
base type, we do not have to override it at all.

greschd commented:
Why do we need to implement our own `__eq__` in the first place? By inheriting from the `namedtuple`, it already has an `__eq__`, no?

The difference I can see to the current implementation is that tuples also compare equal to lists (and maybe other containers) with the same content.

sphuber commented:
> Why do we need to implement our own `__eq__` in the first place? By inheriting from the `namedtuple`, `ExitCode` already has an `__eq__`, no?
> 
> The difference I can see to the current implementation is that tuples also compare equal to lists (and maybe other containers) with the same content.

Uhm, good question :sweat_smile:  I just tried and removed the implementation and the tests I added still pass. That means it does distinguish between type, because the list with the same content does not match (which is what we want I think). So I think you are right and the fix is simply to remove it? Huh

greschd commented:
> 
> That means it does distinguish between type, because the list with the same content does not match (which is what we want I think). So I think you are right and the fix is simply to remove it? Huh

Right, not sure what made me believe it didn't distinguish list / tuple :dizzy_face:

Indeed, simply removing it seems like the easiest fix.

greschd commented on tests/engine/processes/text_exit_code.py:
```diff
@@ -33,6 +33,15 @@ def test_exit_code_construct():
+    serialized = yaml.dump(exit_code)
+    deserialized = yaml.full_load(serialized)
+    assert deserialized == exit_code
+
```
Maybe we should check that the result is still an `ExitCode` instance here?

sphuber commented:
> Nice! Since this is a pretty significant bug, are you planning a hotfix release for this?

We should release it soon I agree, but I think I will hold off a bit longer to see if anything else props up. If nothing is reported over the weekend I will release `1.2.1` on Monday I think. Sound reasonable?
